src/sentry/models/groupsubscription.py

```diff
@@ -94,6 +94,60 @@ def subscribe(self, group, user, reason=GroupSubscriptionReason.unknown):
         except IntegrityError:
             pass
 
+    def subscribe_actor(self, group, actor, reason=GroupSubscriptionReason.unknown):
+        from sentry.models import User, Team
+
+        if isinstance(actor, User):
+            return self.subscribe(group, actor, reason)
+        elif isinstance(actor, Team):
+
+            # subscribe the members of the team
+            team_users_ids = list(actor.member_set.values_list('user_id', flat=True))
+
+            return self.bulk_subscribe(group, team_users_ids, reason)
+        else:
+            raise NotImplementedError('Uknown actor type')
+
+    def bulk_subscribe(self, group, user_ids, reason=GroupSubscriptionReason.unknown):
+        """
+        Subscribe a list of users or users ids to an issue, but only if the users are not explicitly
+        unsubscribed.
+        """
+        # 5 retries for race conditions
+        for _ in range(5):
+
+            existing_subscriptions = set(GroupSubscription.objects.filter(
+                user_id__in=user_ids,
+                group=group,
+                project=group.project,
+            ).values_list('user_id', flat=True))
+
+            subscriptions = []
```

Comment:
nit: since you tend to lean more towards this anyways:

```python
subscriptions = [
    GroupSubscription(...)
    for user_id in user_ids
    if user_id not in existing_subscriptions
]
```

To be clear, I don't care if you change this, just suggesting an alternate style.

Reply:
love it